vllm/entrypoints/openai/serving_completion.py

```diff
@@ -153,7 +153,8 @@ async def create_completion(self, request: CompletionRequest,
                 generators.append(generator)
         except ValueError as e:
             # TODO: Use a vllm-specific Validation Error
-            return self.create_error_response(str(e))
+            msg, status_code = e.args
+            return self.create_error_response(msg, status_code=status_code)
```

Comment:
There is no guarantee the error can be unwrapped like this. How about leaving the existing ValueError handling as is, and add a separate clause on top of it so we can handling `except QueueOverflowError as e`... (same for below). 

You also need a handling for serving_chat